File: Server/StockTKL/src/main/java/com/example/stocktkl/payload/request/OrderRequest.java
package com.example.stocktkl.payload.request;

import com.example.stocktkl.model.enum_class.EOrderDirection;
import com.example.stocktkl.model.enum_class.EOrderType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class OrderRequest {
//    private Long userId;
    private String stockSymbol;
    private EOrderType orderType;
    private EOrderDirection direction;
    private int quantity;
    private BigDecimal price = BigDecimal.valueOf(-1);
}

File: Client/stock_tkl/lib/models/order_request.dart
class OrderRequest {
  final String symbol;
  final String orderType;
  final String direction;
  final int quantity;
  final double price;

  OrderRequest(
      {required this.symbol,
      required this.orderType,
      required this.direction,
      required this.quantity,
      required this.price});
}
